modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,435 @@
+
+						}
+						else if (type == CV_MCT_MEAN_VARIATION)
+						{
+							//to do mean variation
+						}
+						dst1[distV + j] = c;
+						dst2[distV + j] = c2;
+					}
+				}
+			}
+
+		};
+		//!paralel implementation of the center symetric census
+		class parallelSymetricCensus:public ParallelLoopBody
+		{
+		private:
+			uint8_t *image1, *image2;
+			int *dst1, *dst2;
+			int n2, width, height, type;
+		public:
+			parallelSymetricCensus(uint8_t * img1, uint8_t * img2, int w, int h, int k2, int * distance1, int * distance2,const int t) :
+				image1(img1), image2(img2), dst1(distance1), dst2(distance2), n2(k2), width(w), height(h), type(t){}
+
+			virtual void operator()(const cv::Range &r) const {
+
+				for (int i = r.start; i <= r.end ; i++)
+				{
+					int distV = (i)* width;
+					for (int j = n2; j <= width - n2; j++)
+					{
+						int c = 0;
+						int c2 = 0;
+						//the classic center symetric census which compares the curent pixel with its symetric not its center.
+						if (type == CV_CLASSIC_CENTER_SYMETRIC_CENSUS)
+						{
+							for (int ii = -n2; ii < 0; ii++)
+							{
+								int rrWidth = (ii + i) * width;
+								for (int jj = -n2; jj <= +n2; jj++)
+								{
+									if (ii != i || jj != j)
+									{
+										if (image1[(rrWidth + (jj + j))] > image1[((ii * (-1) + i) * width + (-1 * jj) + j)])
+										{
+											c = c + 1;
+										}
+										c = c * 2;
+									}
+									if (ii != i || jj != j)
+									{
+										if (image2[(rrWidth + (jj + j))] > image2[((ii * (-1) + i) * width + (-1 * jj) + j)])
+										{
+											c2 = c2 + 1;
+										}
+										c2 = c2 * 2;
+									}
+								}
+							}
+							for (int jj = -n2; jj < 0; jj++)
+							{
+								if (image1[(i * width + (jj + j))] > image1[(i * width + (-1 * jj) + j)])
+								{
+									c = c + 1;
+								}
+								c = c * 2;
+								if (image2[(i * width + (jj + j))] > image2[(i * width + (-1 * jj) + j)])
+								{
+									c2 = c2 + 1;
+								}
+								c2 = c2 * 2;
+							}
+						}//a modified version of cs census which compares each pixel with its correspondent from
+						//the same distance from the center
+						else if (type == CV_MODIFIED_CENTER_SIMETRIC_CENSUS)
+						{
+							for (int ii = i - n2; ii <= i + 1; ii++)
+							{
+								int rrWidth = ii * width;
+								int rrWidthC = (ii + n2) * width;
+								for (int jj = j - n2; jj <= j + n2; jj += 2)
+								{
+									if (ii != i || jj != j)
+									{
+										if (image1[(rrWidth + jj)] > image1[(rrWidthC + (jj + n2))])
+										{
+											c = c + 1;
+										}
+										c = c * 2;
+									}
+									if (ii != i || jj != j)
+									{
+										if (image2[(rrWidth + jj)] > image2[(rrWidthC + (jj + n2))])
+										{
+											c2 = c2 + 1;
+										}
+										c2 = c2 * 2;
+									}
+								}
+							}
+						}
+						dst1[(distV + j)] = c;
+						dst2[(distV + j)] = c2;
+					}
+				}
+			}
+		};
+		class Descriptor
```

Comment:
I don't think we need another nested namespace for those functions. I suppose it's better to simply put them as plain functions in `cv::stereo`